src/compiler/checker.ts

```diff
@@ -1501,6 +1502,9 @@ namespace ts {
                 }
 
                 if (!excludeGlobals) {
+                    if (name === "globalThis" as __String) {

```

Comment:
See my comment [here](https://github.com/Microsoft/TypeScript/pull/29332/files#r246555756) - this shouldn't be required just by including the symbol in the `builtinGlobals` map.